### PR TITLE
Reimplement wNEAR with near-contract-standards

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
NOTE: Legacy means the contract deployed on `wrap.near` (not NEP-21).

- Updates wNEAR  implementation to use FT standard. Exposes legacy storage method `storage_minimum_balance` which is available on `wrap.near` contract.
- Upgrade `rust-toolchain` to stable Rust `1.51.0`. Rebuild with contract builder.

## Test plan
- Add sim tests to test legacy->current migration
- Verified on testnet

### CODE REVIEW

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-nightly-2020-05-15
+stable-2020-10-08
```
Comment (bowenwang1996) [inline rust-toolchain:1]: if it is stable, why do we even need this file

**Review by bowenwang1996 — COMMENTED** [review]
Did you forget to include some file?

File: rust-toolchain (hunk 1 above)
Comment (evgenykuzyakov) [reply rust-toolchain:1]: It's the one we use for `near-sdk-rs`. Maybe people had old toolchains by default in the past.

**Review by bowenwang1996 — APPROVED** [review]

**Comment by ilblackdragon** [comment]
Let's upgrade this to using near-standard-contracts and push?
We also need to hot swap the current version and update Flux

**Comment by ilblackdragon** [comment]
My suggestion we keep the old storage API for Flux UI to work.

File: w-near-141/Cargo.toml [hunk 2]
```diff
@@ -8,7 +8,10 @@ edition = "2018"
 crate-type = ["cdylib", "rlib"]
 
 [dependencies]
-near-sdk = "2.0.0"
+near-sdk = { git = "https://github.com/near/near-sdk-rs" }
```
Comment (willemneal) [inline w-near-141/Cargo.toml:11]: Should pin a commit until published.

File: w-near/src/lib.rs [hunk 3]
```diff
@@ -1,943 +1,58 @@
...
-            attached_deposit: 0,
-            prepaid_gas: 10u64.pow(18),
-            random_seed: vec![0, 1, 2],
-            is_view: false,
-            output_data_receivers: vec![],
-            epoch_height: 0,
+near_contract_standards::impl_fungible_token_core!(Contract, ft);
+near_contract_standards::impl_fungible_token_storage!(Contract, ft);
+
+impl FungibleTokenMetadataProvider for Contract {
```
Comment (ilblackdragon) [inline w-near/src/lib.rs:46]: #[near_bindgen]

**Review by ilblackdragon — APPROVED** [review]
LGTM with one line missing to expose ft_metadata

**Pull request merged**